### PR TITLE
NETOBSERV-577 Few FLP optimizations

### PR DESCRIPTION
**Pull request by jotak**
- parallel transforms
- minor loki writer optimizations (pre-compute a couple of things)
- add loki benchmark
- add stage measurment metric

### CODE REVIEW

File: pkg/pipeline/transform/transform_network.go [hunk 1]
```diff
@@ -40,9 +41,23 @@ type Network struct {
...
+	entriesChan := make(chan config.GenericMap, len(input))
+	var wg sync.WaitGroup
+	wg.Add(len(input))
+
+	for _, e := range input {
+		go func(entry config.GenericMap) {
+			defer wg.Done()
+			outputEntry := n.TransformEntry(entry)
+			entriesChan <- outputEntry
+		}(e)
```
Comment (mariomac) [inline pkg/pipeline/transform/transform_network.go:53]: I'm afraid that the goroutine creation/management and channel management costs could repel any parallelization improvement. TransformEntry is a short process, and spawning 10K goroutines to hold each of them in parallel could not be effective.

I'd rather divide the work in less tasks carrying a bigger portion of the work. E.g. carry a subslice on each goroutine:

```go
outputEntries := make([]config.GenericMap, len(input))
parallelTasks := runtime.GOMAXPROCS(0) // or halve it, or do user-configurable 
for t := 0; t < parallelTasks ; t++ {
    si := t * len(input) / parallelTasks
    ei := (t + 1) * len(input) / parallelTasks
    go func() {
        defer wg.Done()
        for i, entry := range input[si: ei] {
            outputEntries[i + si] = n.TransformEntry(entry)
        }
    }()
}
```

Comment (mariomac) [reply pkg/pipeline/transform/transform_network.go:53]: ("10K goroutines", thinking in the use case of one of our users)

Comment (jotak) [reply pkg/pipeline/transform/transform_network.go:53]: yeah ... "goroutines are cheap", but maybe not THAT cheap :)

Comment (jotak) [reply pkg/pipeline/transform/transform_network.go:53]: anyway, the need for parallel processing was assuming there was blocking calls on the kube API, but if that's not the case (still unclear to me), we can forget this

**Review by mariomac — APPROVED** [review]

File: pkg/pipeline/write/write_loki_test.go [hunk 2]
```diff
@@ -285,3 +291,57 @@ parameters:
...
+	}
+
+	i := 0
+	for range lokiFlows {
+		i++
+		if i == 100*b.N {
+			close(lokiFlows)
+			break
+		}
+	}
```
Comment (ronensc) [inline pkg/pipeline/write/write_loki_test.go:346]: What's the purpose of this loop?

**Review by ronensc — APPROVED** [review]

File: pkg/pipeline/write/write_loki_test.go (hunk 2 above)
Comment (jotak) [reply pkg/pipeline/write/write_loki_test.go:346]: Reading on the channel is necessary to not block the flows processing. I haven't looked much further in `test.FakeLokiHandler`, I'd agree this code is not super elegant maybe something better can be done, i'll look more into it

**Pull request merged**